apps/bfd-model/bfd-model-dsl-codegen/bfd-model-dsl-codegen-plugin/src/main/java/gov/cms/model/dsl/codegen/plugin/GenerateTransformersFromDslMojo.java

```diff
@@ -0,0 +1,588 @@
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+import gov.cms.model.dsl.codegen.library.DataTransformer;
+import gov.cms.model.dsl.codegen.library.EnumStringExtractor;
+import gov.cms.model.dsl.codegen.library.ExternalTransformation;
+import gov.cms.model.dsl.codegen.plugin.accessor.GrpcGetter;
+import gov.cms.model.dsl.codegen.plugin.accessor.OptionalSetter;
+import gov.cms.model.dsl.codegen.plugin.accessor.RifGetter;
+import gov.cms.model.dsl.codegen.plugin.accessor.StandardSetter;
+import gov.cms.model.dsl.codegen.plugin.model.ArrayBean;
+import gov.cms.model.dsl.codegen.plugin.model.ColumnBean;
+import gov.cms.model.dsl.codegen.plugin.model.ExternalTransformationBean;
+import gov.cms.model.dsl.codegen.plugin.model.MappingBean;
+import gov.cms.model.dsl.codegen.plugin.model.ModelUtil;
+import gov.cms.model.dsl.codegen.plugin.model.RootBean;
+import gov.cms.model.dsl.codegen.plugin.model.TransformationBean;
+import gov.cms.model.dsl.codegen.plugin.transformer.FieldTransformer;
+import gov.cms.model.dsl.codegen.plugin.transformer.TransformerUtil;
+import java.io.File;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import javax.lang.model.element.Modifier;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+
+/**
+ * A Maven Mojo that generates code to copy and transform data from RDA API message objects into JPA
+ * entity objects. The generated transformer class consists of the following parts:
+ *
+ * <ul>
+ *   <li>Private final field declarations section containing one field for each {@link
+ *       FieldTransformer} that requires one.
+ *   <li>Private final field declaration for a lambda function to convert strings into hashed
+ *       strings if one is required by the {@link TransformationBean}.
+ *   <li>Private final field declaration for each {@link ExternalTransformation} lambda function
+ *       required by the {@link TransformationBean}.
+ *   <li>A public constructor that accepts any arguments needed to initialize the private final
+ *       fields.
+ *   <li>A simple public {code transformMessage()} method for each message class defined by the root
+ *       {@link MappingBean}. The simple method accepts just a message class instance and either
+ *       returns a valid entity class instance or throws a {@link
+ *       DataTransformer.TransformationException} if there were any validation errors.
+ *   <li>A more verbose public {@code transformMessage()} method for each message class defined by
```

Comment:
All REALLY cool stuff; would there be any value in providing an internal entity validity/self check (i.e., isValid()) since we'll always get the entity class instance, which may be incomplete. Granted the caller may have different ideas on validity, but the internal processing knows about data transformation processing errors and could provide a short-cut for the caller.

Reply:
Thanks!  Since the validity checks are currently performed during transformation that would require duplicating code between the two mojos.  It might be possible to add just simple checks to the entities like nullability though.